dir.c

```diff
@@ -658,6 +658,38 @@ void clear_exclude_list(struct exclude_list *el)
 	memset(el, 0, sizeof(*el));
 }
 
+static void trim_trailing_comments(char *buf)
```

Comment:
This function is very similar to `trim_trailing_spaces()`. l don't think logic should be put at all costs in one single function, but it's worth thinking if they can be merged somehow.